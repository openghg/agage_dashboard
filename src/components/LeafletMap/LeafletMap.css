.map-header {
  background-color: lightblue;
  font-size: 2vh;
}

.map-container {
  margin: 20px;
  padding: 20px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  background-color: #f4f3ee;
  color: slategray;
}
