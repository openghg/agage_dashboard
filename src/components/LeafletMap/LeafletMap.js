import React from "react";
import { MapContainer, TileLayer, Marker, Popup } from "react-leaflet";
import styles from "./LeafletMap.module.css";

class LeafletMap extends React.Component {
  processSites() {
    const sites = this.props.sites;

    let markers = [];
    for (const [key, value] of Object.entries(sites)) {
      const latitude = value["latitude"];
      const longitude = value["longitude"];

      const locationStr = `${key}, ${latitude}, ${longitude}`;

      const location = [latitude, longitude];

      const marker = (
        <Marker key={locationStr} position={location}>
          <Popup>
            <div className={styles.marker}>
              <div className={styles.markerHeader}>
                {String(key).toUpperCase()}
              </div>
              <div className={styles.markerBody}>{value["long_name"]}</div>
              <div className={styles.markerLocation}>Location: {locationStr}</div>
            </div>
          </Popup>
        </Marker>
      );

      markers.push(marker);
    }

    return markers;
  }

  render() {
    const markers = this.processSites();
    const zoom = this.props.zoom ? this.props.zoom : 5;

    const width = this.props.width ? this.props.width : "60vw";
    const height = this.props.height ? this.props.height : "40vh";

    const style = { width: width, height: height };

    return (
      <div className={styles.container}>
        <MapContainer
          center={this.props.centre}
          zoom={zoom}
          scrollWheelZoom={true}
          style={style}
        >
          <TileLayer
            attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
            url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
          />
          {markers}
        </MapContainer>
      </div>
    );
  }
}

export default LeafletMap;
