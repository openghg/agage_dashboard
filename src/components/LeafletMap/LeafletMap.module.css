.container {
  display: flex;
  flex-direction: row;
  justify-content: center;
  color: slategray;
  width: 100%;
  height: 35vh;
  border-radius: 8px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  padding: 2%;
}

.marker {
  background-color: white;
  display: flex;
  flex-direction: column;
  gap: 1vh;
}

.markerHeader {
  font-size: 1.5em;
  font-weight: 500;
  padding-bottom: 1vh;
  align-content: space-between;
}

.markerTitle {
  font-size: 1em;
  font-weight: 600;
}

.markerBody {
  font-size: 1.2em;
  font-weight: 300;
  padding-bottom: 1vh;
  gap: 1vh;
}

.markerButtons {
  display: flex;
  flex-direction: row;
  padding-top: 2vh;
  gap: 1vw;
}

.markerLocation {
  font-style: normal;
}
