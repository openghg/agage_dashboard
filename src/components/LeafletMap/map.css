.mapid {
  width: 100%;
  height: 100%;
  overflow-x: scroll;
  overflow-y: hidden;
}

.leaflet-container {
  width: 100%;
  height: 100%;
}

