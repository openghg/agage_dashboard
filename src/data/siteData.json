{
  "TMB": {
    "long_name": "Thames Barrier",
    "latitude": 51.497,
    "longitude": 0.037,
    "measurements": {"2021-01-01": "66", "2021-01-03": "99"}
  },
  "BTT": {
    "long_name": "BT Tower",
    "latitude": 51.5215,
    "longitude": -0.1389,
    "measurements": {"2021-01-01": "66", "2021-01-03": "99"}
  },
  "NPL": {
    "long_name": "National Physical Laboratory",
    "latitude": 51.4241,
    "longitude": -0.3487,
    "measurements": {"2021-01-01": "66", "2021-01-03": "99"}
  },
  "UOW": {
    "long_name": "University of Westminster, Harrow Campus",
    "latitude": 51.5776,
    "longitude": -0.3251,
    "measurements": {"2021-01-01": "66", "2021-01-03": "99"}
  },
  "STJ": {
    "long_name": "St Jude on the Hill",
    "latitude": 51.5802,
    "longitude": -0.1899,
    "measurements": {"2021-01-01": "66", "2021-01-03": "99"}
  },
  "HFT": {
    "long_name": "Highfield Tower",
    "latitude": 51.608,
    "longitude": 0.1745,
    "measurements": {"2021-01-01": "66", "2021-01-03": "99"}
  },
  "SMA": {
    "long_name": "St Mary's",
    "latitude": 51.5044,
    "longitude": -0.3056,
    "measurements": {"2021-01-01": "66", "2021-01-03": "99"}
  },
  "RHO": {
    "long_name": "Royal Holloway",
    "latitude": 51.427582,
    "longitude": -0.56278,
    "measurements": {"2021-01-01": "66", "2021-01-03": "99"}
  }
}
