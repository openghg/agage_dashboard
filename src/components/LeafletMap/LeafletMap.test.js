import { render } from "@testing-library/react";
import React from "react";

import LeafletMap from "./LeafletMap";
import siteData from "../../data/siteMetadata.json";

describe("Test LineChart", () => {
  test("Check correct chart rendered", () => {
    const { asFragment } = render(
      <LeafletMap sites={{ TMB: siteData["TMB"] }} centre={[51.5, -0.0482]} zoom={10} width={"75vw"} height={"65vh"} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
