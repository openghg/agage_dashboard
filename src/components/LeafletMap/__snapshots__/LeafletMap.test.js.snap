// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LineChart Check correct chart rendered 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="leaflet-container leaflet-grab leaflet-touch-drag"
      style="width: 75vw; height: 65vh; position: relative;"
      tabindex="0"
    >
      <div
        class="leaflet-pane leaflet-map-pane"
        style="left: 0px; top: 0px;"
      >
        <div
          class="leaflet-pane leaflet-tile-pane"
        >
          <div
            class="leaflet-layer "
            style="z-index: 1;"
          >
            <div
              class="leaflet-tile-container leaflet-zoom-animated"
              style="z-index: 18; left: 0px; top: 0px;"
            >
              <img
                alt=""
                class="leaflet-tile"
                role="presentation"
                src="https://c.tile.openstreetmap.org/10/511/340.png"
                style="width: 256px; height: 256px; left: -221px; top: -138px;"
              />
            </div>
          </div>
        </div>
        <div
          class="leaflet-pane leaflet-shadow-pane"
        >
          <img
            alt=""
            class="leaflet-marker-shadow leaflet-zoom-hide"
            src="marker-shadow.png"
            style="margin-left: -12px; margin-top: -41px; width: 41px; height: 41px; left: 62px; top: 4px;"
          />
        </div>
        <div
          class="leaflet-pane leaflet-overlay-pane"
        />
        <div
          class="leaflet-pane leaflet-marker-pane"
        >
          <img
            alt=""
            class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
            src="marker-icon.png"
            style="margin-left: -12px; margin-top: -41px; width: 25px; height: 41px; left: 62px; top: 4px; z-index: 4;"
            tabindex="0"
          />
        </div>
        <div
          class="leaflet-pane leaflet-tooltip-pane"
        />
        <div
          class="leaflet-pane leaflet-popup-pane"
        />
      </div>
      <div
        class="leaflet-control-container"
      >
        <div
          class="leaflet-top leaflet-left"
        >
          <div
            class="leaflet-control-zoom leaflet-bar leaflet-control"
          >
            <a
              aria-label="Zoom in"
              class="leaflet-control-zoom-in"
              href="#"
              role="button"
              title="Zoom in"
            >
              +
            </a>
            <a
              aria-label="Zoom out"
              class="leaflet-control-zoom-out"
              href="#"
              role="button"
              title="Zoom out"
            >
              −
            </a>
          </div>
        </div>
        <div
          class="leaflet-top leaflet-right"
        />
        <div
          class="leaflet-bottom leaflet-left"
        />
        <div
          class="leaflet-bottom leaflet-right"
        >
          <div
            class="leaflet-control-attribution leaflet-control"
          >
            <a
              href="https://leafletjs.com"
              title="A JS library for interactive maps"
            >
              Leaflet
            </a>
             | © 
            <a
              href="http://osm.org/copyright"
            >
              OpenStreetMap
            </a>
             contributors
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
